plugins/module_utils/mh/base.py

```diff
@@ -63,3 +63,10 @@ def run(self):
         if 'failed' not in output:
             output['failed'] = False
         self.module.exit_json(changed=self.has_changed(), **output)
+
+    @classmethod
+    def execute(cls, module=None):
```

Comment:
I think that adding the `module` arg into the signature is premature. If one wants to customize it, they could change this method via inheritance. What is the use-case here? Dependency injection for testing?

Reply:
If, for any reason, the module developer needs to have control of how the `AnsibleModule` is created, they can do so. There is absolutely no need for us to impose the module instantiation within `ModuleHelper`.

Reply:
> If, for any reason

This sounds like a very definition of "premature" to me...

If we say that it's necessary to replace the need for monkey-patching in tests with dependency injection, it'd be a more valid reason to me.

Reply:
> > If, for any reason
> 
> This sounds like a very definition of "premature" to me...

I can think of a scenario where we might have different module classes (based on MH) available and the one that gets used is dependent on some incoming parameter value. In taht case, one needs to instantiate the module *first*, to be able to read `module.params['deciding_factor']`, then choose which MH-based class to use.

Reply:
Could you give me a more comprehensive example?

Reply:
I'm +-0 of having the `module` argument: +1 because `__init__` also has that argument; and -1 because this convenience function is more for the simple case (just run the module), if you want to provide the `AnsibleModule` instantiation yourself, you already have a few more lines and you can also instantiate the module's class itself and call it's `run()` method.

Reply:
I would go as a +0.5 then. Keeping the consistency of the interfaces there feels more important than this alleged need to simplify this convenience method. Add to that the fact that the only parameter is optional, so it can (and will mostly) be called without any args at all, I find it hard to support that argument.

Reply:
Ok, then let's keep it.